### ISSUE TITLE
sql: users can connect even if they don't exist in insecure mode

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#20717]
```
[11:12]% cockroach sql --insecure -u nosuchuser
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
#
# Client version: CockroachDB CCL v1.2-alpha.20170901-2084-ge457f9a19 (darwin amd64, built 2017/11/30 14:50:40, go1.9.1)
# Server version: CockroachDB CCL v1.2-alpha.20171211-123-g6172ba8d6 (darwin amd64, built 2017/12/11 22:32:52, go1.9.1)
# Cluster ID: b6057ca9-c5ef-47f4-800c-60ebc3749eff
#
# Enter \? for a brief introduction.
#
nosuchuser@:26257/> select 1;
+---+
| 1 |
+---+
| 1 |
+---+
(1 row)
```

This problem is not present in secure mode:

```
[11:14]% ./cockroach sql -u nosuchuser
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
#
Enter password:
Error: pq: user nosuchuser does not exist
Failed running "sql"
```

**Comment by mberhault** [comment]
@vivek for triage.

Given that anyone can login as root in insecure mode, that's probably not a very high-priority issue.

**Comment by mberhault** [comment]
True. But existing sessions are different from login checks. Let's keep #20718 for the former and leave this issue for user existence checks at login time in insecure mode.

**Comment by vivekmenezes** [comment]
anyone think this is something we need to fix for 2.0?
